core/node/da_clients/src/eigen/memstore.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::{Duration, Instant},
+};
+
+use anyhow::Error;
+use rand::{rngs::OsRng, Rng, RngCore};
+use sha3::{Digest, Keccak256};
+use tokio::time::interval;
+use zksync_config::configs::da_client::eigen::MemStoreConfig;
+use zksync_da_client::types::{DAError, InclusionData};
+
+use super::blob_info::{self, BlobInfo};
+
+#[derive(Debug, PartialEq)]
+pub enum MemStoreError {
+    BlobToLarge,
+    BlobAlreadyExists,
+    IncorrectCommitment,
+    #[cfg(test)]
+    BlobNotFound,
+}
+
+impl From<MemStoreError> for Error {
+    fn from(val: MemStoreError) -> Self {
+        match val {
+            MemStoreError::BlobToLarge => Error::msg("Blob too large"),
+            MemStoreError::BlobAlreadyExists => Error::msg("Blob already exists"),
+            MemStoreError::IncorrectCommitment => Error::msg("Incorrect commitment"),
+            #[cfg(test)]
+            MemStoreError::BlobNotFound => Error::msg("Blob not found"),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct MemStoreData {
+    store: HashMap<String, Vec<u8>>,
+    key_starts: HashMap<String, Instant>,
+}
+
+/// This struct represents a memory store for blobs.
+/// It should be used for testing purposes only.
+#[derive(Clone, Debug)]
+pub struct MemStore {
```

Comment:
we have a NoDA client for those who don't want to dispatch blobs anywhere and a full Eigen impl for those who do, what are the use cases for memstore?

Reply:
ok, we will remove it. It was added originally since it served for testing purposes.

Reply:
Removed here https://github.com/lambdaclass/zksync-era/pull/346